plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixSplitManager.java

```diff
@@ -58,6 +59,7 @@
         implements ConnectorSplitManager
 {
     private static final Logger log = Logger.get(PhoenixSplitManager.class);
+    private static final int MAX_SCANS_PER_SPLIT = 20;
```

Comment:
Justification for this value in code comment (and commit description) would be helpful.

IIUC this mechanism reduces the number of scans being performed on Phoenix by grouping some of them within a single split?

Reply:
Phoenix uses two metrics to split queries into multiple HBase scans:
1. HBase regions. Each region is handled by a separate (set of) scan(s)
2. Since HBase regions can get large, Phoenix keeps additional metrics: Guideposts, where it records the value of a PK (which are ordered) every N bytes (default 300MB). And if enabled each Guideposts that falls into the range of keys touched by a query turns into a separate scan.

So with this patch we find a compromise. A scan per region leaves compute resources unused, and a scan per Guidepost is overkill especially for larger tables.

I'll put something in the comment.